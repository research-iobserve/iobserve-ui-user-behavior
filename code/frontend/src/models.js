// @flow

import { List, Map } from "immutable";
import Record from "./types";
import type {
  ApplicationT,
  ApplicationStateT,
  PageT,
  PageStateT,
  VisitT,
  VisitStateT,
  SettingsStateT,
  LayoutT,
  StateT
} from "./types";

/**
 * Application
 */
const applicationDefaults: ApplicationT = {
  id: 0,
  name: "",
  description: "",
  custom: false
};
export class Application extends Record(applicationDefaults) {
  id: number;
  name: string;
  description: string;
  custom: boolean;
}

/**
 * Page
 */
const pageDefaults: PageT = {
  id: 0,
  name: "",
  extra: Map()
};
export class Page extends Record(pageDefaults) {
  id: number;
  name: string;
  extra: Map<string, string>;
}

/**
 * Visit
 */
const visitDefaults: VisitT = {
  id: 0,
  count: 0,
  start: new Page(),
  end: new Page(),
  action: ""
};
export class Visit extends Record(visitDefaults) {
  id: number;
  count: number;
  start: Page;
  end: Page;
  action: string;
}

/**
 * ApplicationState
 */
const applicationStateDefaults: ApplicationStateT = {
  applications: List()
};
export class ApplicationState extends Record(applicationStateDefaults) {
  applications: List<Application>;
}

/**
 * PageState
 */
const pageStateDefaults: PageStateT = {
  pages: Map()
};
export class PageState extends Record(pageStateDefaults) {
  pages: Map<number, List<Page>>;
}

/**
 * VisitState
 */
const visitStateDefaults: VisitStateT = {
  visits: Map()
};
export class VisitState extends Record(visitStateDefaults) {
  visits: Map<number, List<Visit>>;
}

/**
 * SettingsState
 */
const settingsStateDefaults: SettingsStateT = {
  layout: "breadthfirst"
};
export class SettingsState extends Record(settingsStateDefaults) {
  layout: LayoutT;
}

/**
 * State
 */
const stateDefaults: StateT = {
  application: new ApplicationState(),
  page: new PageState(),
  visit: new VisitState(),
  settings: new SettingsState()
};
export class State extends Record(stateDefaults) {
  application: ApplicationState;
  page: PageState;
  visit: VisitState;
  settings: SettingsState;
}
