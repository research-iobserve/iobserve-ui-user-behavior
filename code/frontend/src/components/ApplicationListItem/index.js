// @flow

import React from "react";
import { Link } from "react-router";
import { Card, CardBlock, CardTitle, CardLink, CardText } from "reactstrap";
import styled from "styled-components";

import { Application } from "../../models";

const StyledCard = styled(Card)`
  width: 30%;
`;

function ApplicationListItem(
  { application, isEditor }: { application: Application, isEditor: boolean }
) {
  return (
    <StyledCard>
      <CardBlock>
        <CardTitle>{application.get("name")}</CardTitle>
        <CardText>{application.get("description")}</CardText>
      </CardBlock>
      <CardBlock>
        <CardLink tag={Link} to={`/applications/${application.get("id")}`}>
          Inspect
        </CardLink>
        {isEditor
          ? <CardLink tag={Link} to={`/editor/${application.get("id")}`}>
            Edit
          </CardLink>
        : null}
      </CardBlock>
    </StyledCard>
  );
}

export default ApplicationListItem;
