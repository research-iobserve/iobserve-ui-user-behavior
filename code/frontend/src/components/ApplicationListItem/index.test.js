// @flow

import React from "react";
import renderer from "react-test-renderer";

import { Application } from "../../models";
import ApplicationListItem from "./index";

const application = new Application({
  id: 0,
  name: "TestApplication",
  description: "Test ..."
});

describe("<ApplicationListItem />", () => {
  it("should render a application", () => {
    const tree = renderer
      .create(
        <ApplicationListItem application={application} isEditor={false} />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
