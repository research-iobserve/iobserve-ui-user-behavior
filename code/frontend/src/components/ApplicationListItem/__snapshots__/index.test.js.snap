exports[`<ApplicationListItem /> should render a application 1`] = `
<div
  className="bTmpoo card">
  <div
    className="card-block">
    <h4
      className="card-title">
      TestApplication
    </h4>
    <p
      className="card-text">
      Test ...
    </p>
  </div>
  <div
    className="card-block">
    <a
      className="card-link"
      onClick={[Function]}
      style={Object {}}>
      Inspect
    </a>
  </div>
</div>
`;
